### PR TITLE
feat: add command ':Neotest'

### PR DESCRIPTION
**Pull request by MunifTanjim**
Add initial implementation of the `:Neotest` command with completion support.

Instead of adding a lot of command, it's cleaner to have a single one.

The structure will be like this:

`:Neotest [cmd] [subcmd] [...]`

Most of the functionality that I could find are exposed with this command.

---

- Resolve https://github.com/nvim-neotest/neotest/issues/241
- Resolve https://github.com/nvim-neotest/neotest/issues/252
- https://github.com/nvim-neotest/neotest/discussions/227

### CODE REVIEW

File: plugin/neotest.lua [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    "toggle",
+  },
+}
+
+local opt_completion_store = {
+  adapter = function(arg)
+    local part = arg:gsub("^adapter=", "")
+    return vim.tbl_filter(function(cmd)
+      return not not string.find(cmd, "^" .. part)
+    end, require("neotest").run.adapters())
```
Comment (rcarriga) [inline plugin/neotest.lua:42]: Use `require("neotest").state.adapter_ids()` instead, `neotest.run.adapters` is deprecated

**Review by rcarriga — CHANGES_REQUESTED** [review]
This is brilliant, thank you very much for the PR! :smile: Just one small comment, then LGTM

File: plugin/neotest.lua (hunk 1 above)
Comment (MunifTanjim) [reply plugin/neotest.lua]: Updated

**Comment by rcarriga** [comment]
Thanks again for this, very nice to have :grin:

**Comment by Zeioth** [comment]
@MunifTanjim there is some equivalent to...?

```
require('neotest').run.run({ strategy = 'dap' })
```

**Comment by MunifTanjim** [comment]
@Zeioth that would be `:Neotest run strategy=dap`

**Pull request merged**